### PR TITLE
Per-buffer blending

### PR DESCRIPTION
**Pull request by IMS212**
yeah

### CODE REVIEW

File: src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java [hunk 1]
```diff
@@ -192,8 +198,44 @@ public ShaderProperties(String contents, ShaderPackOptions shaderPackOptions, It
...
+						bufferBlendOverrides.computeIfAbsent(parts[0], list -> new ArrayList<>()).add(new BufferBlendOverride(index, null));
+						return;
+					}
+
+					String[] modeArray = value.split(" ");
+					int[] modes = new int[4];
+
+					int i = 0;
+					for (String modeName : modeArray) {
+						modes[i] = BlendModeFunction.fromString(modeName).get().getGlId();
```
Comment (MahirZukic) [inline src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java:233]: We could get an `ArrayIndexOutOfBounds` Exception here due to the 229th line.
Shouldn't that be?
`int[] modes = new int[modeArray.length];`

**Review by MahirZukic — CHANGES_REQUESTED** [review]
We could get an ArrayIndexOutOfBounds Exception like I already commented.

File: src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java:233]: Thanks for pointing this out, fixed

**Comment by IMS212** [comment]
Will merge soon if there are no complaints.

**Pull request merged**